### PR TITLE
[Turnstile] Add guide to use Turnstile+Terraform

### PR DESCRIPTION
**Pull request by punkeel**
Let's wait for the terraform version 4.7.0

**Rendered: https://maxime-turnstile-tf.cloudflare-docs-7ou.pages.dev/turnstile/reference/terraform/**

### CODE REVIEW

**Comment by Cyb3r-Jak3** [comment]
Don't know if it makes sense to make a note somewhere that `id` == `sitekey`. There are subtle references in the example configuration already.

**Comment by WalshyDev** [comment]
Been meaning to make a page like this for Pages, totally gonna steal when this is merged

**Comment by punkeel** [comment]
> Don't know if it makes sense to make a note somewhere that `id` == `sitekey`. There are subtle references in the example configuration already.

Done, added a note in Step 2.

File: content/turnstile/reference/terraform.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+```
+
+### Step 2: Create a Turnstile widget
+
+Here is an example configuration:
+
+```tf
+---
+header: cloudflare.tf
+highlight: [15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32]
```
Comment (Cherry) [inline content/turnstile/reference/terraform.md]:
```suggestion
highlight: 15-32
```
(will need a rebase from `main` though)

Comment (punkeel) [reply content/turnstile/reference/terraform.md]: done, thanks for implementing that!

File: content/turnstile/reference/terraform.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+pcx_content_type: how-to
+title: Terraform
+weight: 10
+---
+
+# Manage Turnstile with Terraform
+
+{{<Aside type="note"header="Requirements">}}
+
+This guide assumes you already have the [Terraform](https://developer.hashicorp.com/terraform/tutorials/certification-associate-tutorials/install-cli) command installed on your machine.
```
Comment (patriciasantaana) [inline content/turnstile/reference/terraform.md]:
```suggestion
This guide assumes that you have the [Terraform](https://developer.hashicorp.com/terraform/tutorials/certification-associate-tutorials/install-cli) command installed on your machine.
```

File: content/turnstile/reference/terraform.md [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+{{</Aside>}}
+
+[Terraform](https://developer.hashicorp.com/terraform/tutorials/certification-associate-tutorials/install-cli) is a tool for building, changing, and versioning infrastructure, and provides components and documentation for building [Cloudflare resources](https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs). Listed below are examples to help you get started with Turnstile using Terraform. For a more generalized guide on configuring Cloudflare and Terraform, visit our [Getting Started with Terraform and Cloudflare](https://blog.cloudflare.com/getting-started-with-terraform-and-cloudflare-part-1/) blog post.
+
+## Create a Turnstile widget with Terraform
+
+### Step 1: Create an API Token
```
Comment (patriciasantaana) [inline content/turnstile/reference/terraform.md]:
```suggestion
### Create an API token
```

File: content/turnstile/reference/terraform.md [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+
+Create an [API Token](/fundamentals/api/get-started/create-token/) with the **Account > Turnstile > Edit** permission. Next, we need to export this secret in our environment variables:
```
Comment (patriciasantaana) [inline content/turnstile/reference/terraform.md]:
```suggestion
Create an [API Token](/fundamentals/api/get-started/create-token/) with the **Account > Turnstile > Edit** permission. Next, you need to export this secret in our environment variables:
```

File: content/turnstile/reference/terraform.md [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+
+```sh
+$ export CLOUDFLARE_API_TOKEN=<YOUR_API_TOKEN>
+```
+
+### Step 2: Create a Turnstile widget
```
Comment (patriciasantaana) [inline content/turnstile/reference/terraform.md]:
```suggestion
### Create a Turnstile widget
```

File: content/turnstile/reference/terraform.md [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+    description = "Secret key"
+    value       = cloudflare_turnstile_widget.example.secret
+    sensitive   = true
+}
+
+```
+
+{{<Aside type="note">}}
+
+Note: the `id` field in the `cloudflare_turnstile_widget.example` resource is your Turnstile widget's sitekey.
```
Comment (patriciasantaana) [inline content/turnstile/reference/terraform.md]:
```suggestion
The `id` field in the `cloudflare_turnstile_widget.example` resource is your Turnstile widget's sitekey.
```

File: content/turnstile/reference/terraform.md [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+
+{{</Aside>}}
+
+
```
Comment (patriciasantaana) [inline content/turnstile/reference/terraform.md:78]:
```suggestion
```

File: content/turnstile/reference/terraform.md [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+### Step 3: Initialize Terraform and the Cloudflare provider
```
Comment (patriciasantaana) [inline content/turnstile/reference/terraform.md]:
```suggestion
### Initialize Terraform and the Cloudflare provider
```

File: content/turnstile/reference/terraform.md [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+should now work.
+
+If you ever set or change modules or backend configuration for Terraform,
+rerun this command to reinitialize your working directory. If you forget, other
+commands will detect it and remind you to do so if necessary.
+```
+
+
+
+### Step 4: Review the Terraform plan
```
Comment (patriciasantaana) [inline content/turnstile/reference/terraform.md]:
```suggestion
### Review the Terraform plan
```

File: content/turnstile/reference/terraform.md [hunk 10]
```diff
@@ -0,0 +1,210 @@
...
+{{<Aside type="note">}}
+
+Note: the `id` field in the `cloudflare_turnstile_widget.example` resource is your Turnstile widget's sitekey.
+
+{{</Aside>}}
+
+
+### Step 3: Initialize Terraform and the Cloudflare provider
+
+Run the command `terraform init` to set up your Terraform working directory, enabling it to interact with Cloudflare services. This process involves downloading the required provider plugins, establishing backend storage for your state files, and creating a local .terraform directory to store configuration data.
```
Comment (patriciasantaana) [inline content/turnstile/reference/terraform.md]:
```suggestion
Run the command `terraform init` to set up your Terraform working directory, enabling it to interact with Cloudflare services. This process involves downloading the required provider plugins, establishing backend storage for your state files, and creating a local `.terraform` directory to store configuration data.
```

File: content/turnstile/reference/terraform.md [hunk 11]
```diff
@@ -0,0 +1,210 @@
...
+    + turnstile_example_secretkey = (sensitive value)
+    + turnstile_example_sitekey   = (known after apply)
+
+───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
+
+Note: You didn't use the -out option to save this plan, so Terraform can't guarantee to take exactly these actions if you run "terraform apply" now.
+```
+
+
+### Step 5: Apply the Terraform changes
```
Comment (patriciasantaana) [inline content/turnstile/reference/terraform.md]:
```suggestion
### Apply the Terraform changes
```

File: content/turnstile/reference/terraform.md [hunk 12]
```diff
@@ -0,0 +1,210 @@
...
+$ terraform output turnstile_example_secretkey
+"0x4AAAAAAAEe4xWueFq9yX8ypjlimbk1Db4"
+```
+
+## Going further
+These are the basics to get up and running with Turnstile and Terraform.
+Follow the [Get started](/turnstile/get-started/) guide to start using your
+Turnstile sitekey on your website.
+
+For advanced usage, refer to our [Terraform resource documentation](https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs/resources/turnstile_widget).
```
Comment (patriciasantaana) [inline content/turnstile/reference/terraform.md]:
```suggestion
{{<aside type="note">}}
For advanced usage, refer to our [Terraform resource documentation](https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs/resources/turnstile_widget).
{{</aside>}}
```

File: content/turnstile/reference/terraform.md [hunk 13]
```diff
@@ -0,0 +1,210 @@
...
+
+───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
+
+Note: You didn't use the -out option to save this plan, so Terraform can't guarantee to take exactly these actions if you run "terraform apply" now.
+```
+
+
+### Step 5: Apply the Terraform changes
+
+Apply these changes using the `terraform apply` command, once they look accurate and you're comfortable moving forward:
```
Comment (patriciasantaana) [inline content/turnstile/reference/terraform.md]:
```suggestion
Once the changes look accurate and you are comfortable moving forward, apply them using the `terraform apply` command:
```

File: content/turnstile/reference/terraform.md [hunk 14]
```diff
@@ -0,0 +1,210 @@
...
+
+Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
+
+Outputs:
+
+turnstile_example_secretkey = <sensitive>
+turnstile_example_sitekey = "0x4AAAAAAAEe4wQdBshJxBeK"
+```
+
+We have successfuly created a Turnstile widget. Go to [Cloudflare dashboard](https://dash.cloudflare.com/?to=/:account/turnstile) to view its configuration and analytics in a user-friendly interface.
```
Comment (patriciasantaana) [inline content/turnstile/reference/terraform.md]:
```suggestion
You have successfuly created a Turnstile widget. Go to [Cloudflare dashboard](https://dash.cloudflare.com/?to=/:account/turnstile) to view its configuration and analytics in a user-friendly interface.
```

File: content/turnstile/reference/terraform.md [hunk 15]
```diff
@@ -0,0 +1,210 @@
...
+
+### Step 6: Retrieve the secret key
```
Comment (patriciasantaana) [inline content/turnstile/reference/terraform.md]:
```suggestion
### Retrieve the secret key
```

**Review by patriciasantaana — APPROVED** [review]

**Review by worenga — APPROVED** [review]

**Comment by punkeel** [comment]
https://github.com/cloudflare/terraform-provider-cloudflare/releases/tag/v4.7.0 is in, this can now be merged 👍

**Pull request merged**